app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/StartController.scala

```diff
@@ -72,8 +76,24 @@ class StartController @Inject()(
                   case Some(knownFacts) => sessionStoreService.cacheKnownFactsResult(knownFacts)
                   case None             => Future.successful(())
                 }
+            (wasPartiallySubscribed, optAllocatedArn) <- knownFactsResultOpt match {
+                                                          case Some(knownFact) =>
+                                                            subscriptionService.completePartialSubscription(
+                                                              CompletePartialSubscriptionBody(
+                                                                knownFact.utr,
+                                                                SubscriptionRequestKnownFacts(knownFact.postcode)))
+                                                          case None => Future successful (false, None)
+                                                        }
           } yield {
             knownFactsResultOpt match {
+              case Some(_) if wasPartiallySubscribed => {
+                val obtainedArn = optAllocatedArn
+                  .getOrElse(
```

Comment:
That this case can even be considered might be symptomatic of a strange interface to the `completePartialSubscription` function.
I wonder if `completePartialSubscription` should just return an `Option[Arn]` - so if it's a Some then they were partially subscribed and now completely subscribed, and if it's a None they were and still are unsubscribed. 
That might help clear up some fiddly bits around calling the function and dealing with its result.